Modules/Core/Common/itk-module.cmake

```diff
@@ -13,6 +13,7 @@ endif()
 itk_module(ITKCommon
   ENABLE_SHARED
   DEPENDS
+    ITKEigen3
```

Comment:
It will compile, but it will generate zillions of warnings `-Wshadow` in gcc.
Linking to the interface library will use `target_include_directories(xxx SYSTEM)` which uses `-isystem` instead of `-I` avoiding reporting the warnings. These false positives have been reported upstream, and seem related with a gcc bug.

The other option is disable that warning globally when including the itk_eigen.h header.
```
#if defined(__GNUC__)
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wshadow"
    #include ITK_EIGEN(X)
#endif
```

The problem with that is that we cannot `#pragma GCC diagnostic pop` in that header (it won't solve the warnings). So that warning will be disabled for that compilation unit.

The other options is to keep it in the PUBLIC DEPENDS like it is now.

Reply:
Cool, thanks for the explanation.